src/neo/SmartContract/InteropService.Contract.cs

```diff
@@ -56,44 +55,39 @@ private static bool Contract_Update(ApplicationEngine engine)
                 var manifest = engine.CurrentContext.EvaluationStack.Pop().GetString();
                 if (manifest.Length > ContractManifest.MaxLength) return false;
 
-                var contract = engine.Snapshot.Contracts.TryGet(engine.CurrentScriptHash);
-                if (contract is null) return false;
-
+                var oldcontract = engine.Snapshot.Contracts.TryGet(engine.CurrentScriptHash);
+                if (oldcontract is null || oldcontract.HasUpgraded) return false;
```

Comment:
It's only possible to be upgraded only once?

Reply:
Contract A can be upgraded to Contract B,then Contract B can be upgraded to Contract C.
But Contract A can't be upgraded again if it has been upgraded.

Reply:
HasUpgraded to WasUpgraded?

Reply:
OK,`WasUpgraded` is a good name.I will adjust